File: java/spring-hibernate/src/main/java/com/sprint/sms/api/domain/RoleMenu.java
package com.sprint.sms.api.domain;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.persistence.UniqueConstraint;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import com.ideahut.common2.annotation.IdhFormatter;
import com.ideahut.shared2.domain.base.StringIdDomain;

@Entity
@Table(name = "t_role_menu", uniqueConstraints = {@UniqueConstraint(columnNames = {"f_role", "f_menu"})})
@IdhFormatter
public class RoleMenu extends StringIdDomain {
	/**
	 * 
	 */
	private static final long serialVersionUID = -459517068833168887L;
	
	private Role role;
	
	private Menu menu;
	
	private String action;
	
	private Set<String> actionAsSet;

	@ManyToOne(targetEntity = Role.class, fetch = FetchType.EAGER)
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JoinColumn(name = "f_role", nullable = false)
	@IdhFormatter(field = Role.ID)
	public Role getRole() {
		return role;
	}

	public void setRole(Role role) {
		this.role = role;
	}
	
	@ManyToOne(targetEntity = Menu.class, fetch = FetchType.EAGER)
	@OnDelete(action = OnDeleteAction.CASCADE)
	@JoinColumn(name = "f_menu", nullable = false)
	@IdhFormatter(field = Menu.ID)
	public Menu getMenu() {
		return menu;
	}

	public void setMenu(Menu menu) {
		this.menu = menu;
	}
	
	@Column(name = "action")
	@IdhFormatter
	public String getAction() {
		return action;
	}

	public void setAction(String action) {
		this.action = action;
	}

	@Transient
	public Set<String> getActionAsSet() {
		if (actionAsSet != null) {
			return actionAsSet;
		}
		Set<String> set = new HashSet<String>();
		if (action != null) {
			String[] ss = action.split(",");
			for (String s : ss) {
				s = s.trim();
				if (s.length() != 0) {
					set.add(s);
				}
			}
		}
		actionAsSet = Collections.unmodifiableSet(set);
		return actionAsSet;
	}
	
}

File: php/slim-doctrine/app/src/com/sprint/sms/api/domain/RoleMenu.php
<?php
namespace App\com\sprint\sms\api\domain;

use Doctrine\ORM\Mapping as ORM;

use \App\com\sprint\sms\api\domain\Role;
use \App\com\sprint\sms\api\domain\Menu;
use \App\com\sprint\sms\api\support\FormatSupport;

/**
 * @ORM\Entity
 * @ORM\Table(name = "t_role_menu", uniqueConstraints = {@ORM\UniqueConstraint(name = "role_menu_idx", columns = {"f_role", "f_menu"})}) 
 */
class RoleMenu implements FormatSupport
{

	/**
     * @ORM\Column(name="ID_", type = "string")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy = "UUID")
     */
	private $id;
	
	/**
	 * @ORM\ManyToOne(targetEntity = "Role", fetch = "EAGER")
	 * @ORM\JoinColumn(name = "f_role", referencedColumnName = "ID_", nullable = false, onDelete="CASCADE")
	 */ 
    private $role;
	
	/**
	 * @ORM\ManyToOne(targetEntity = "Menu", fetch = "EAGER")
	 * @ORM\JoinColumn(name = "f_menu", referencedColumnName = "ID_", nullable = false, onDelete="CASCADE")
	 */ 
    private $menu;
	
	/**
     * @ORM\Column(name = "action", type = "string", length = 150, nullable = false)
     */
    private $action;
	
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}
	
	public function setRole(Role $role)
    {
        $this->role = $role;
    }
	
    public function getRole()
    {
        return $this->role;
    }	
	
	public function setMenu(Menu $menu)
    {
        $this->menu = $menu;
    }
	
    public function getMenu()
    {
        return $this->menu;
    }
	
	public function setAction($action) {
		$this->action = $action;
	}
	
	public function getAction() {
		return $this->action;
	}
	
	
	
	private $actionAsSet;
	
	public function getActionAsSet() {
		if ($this->actionAsSet != null) {
			return $this->actionAsSet;
		}
		$this->actionAsSet = array();
		if ($this->action === null) {
			return $this->actionAsSet;
		}
		$split = explode(",", $this->action);
		$count = count($split);
		for ($i = 0; $i < $count; $i++) {
			$act = trim($split[$i]);
			if ($act === "") {
				continue;
			}
			array_push($this->actionAsSet, $act);
		}
		return $this->actionAsSet;
	}
	
	public function toFormatObject() {
		$o = array();
		$o["id"] = $this->id;
		$o["role"] = isset($this->role) ? array("id" => $this->role->getId()) : null;
		$o["menu"] = isset($this->menu) ? array("id" => $this->menu->getId()) : null;
		$o["action"] = $this->action;
		return $o;
	}
}